gateway/gateway.go

```diff
@@ -58,10 +58,12 @@ type SessionStartLimit struct {
 func URL() (string, error) {
 	var g BotData
 
-	return g.URL, httputil.NewClient().RequestJSON(
-		&g, "GET",
-		EndpointGateway,
-	)
+	c := httputil.NewClient()
+	if err := c.RequestJSON(&g, "GET", EndpointGateway); err != nil {
+		return "", err
+	}
+
+	return g.URL, nil
```

Comment:
What exactly does splitting this up accomplish? Seeing as `BotData` is not a pointer, `g.URL` would be `""` in an error scenario as well, so this revision should perform exactly the same as the previous version.

Reply:
It does perform exactly the same. I only did this to make the line shorter and to make the code more obvious.

Reply:
Ah, alright!